src/config.rs

```diff
@@ -748,4 +756,106 @@ impl TiKvConfig {
             self.raft_store.region_split_size = default_raft_store.region_split_size;
         }
     }
+
+    pub fn check_critical_cfg_with(&self, last_cfg: &Self) -> Result<(), Box<Error>> {
+        if last_cfg.rocksdb.wal_dir != self.rocksdb.wal_dir {
+            return Err(format!(
+                "db wal_dir have been changed, former db wal_dir is {}, \
+                 current db wal_dir is {}, please guarantee all data wal log \
+                 have been moved to destination directory.",
+                last_cfg.rocksdb.wal_dir, self.rocksdb.wal_dir
+            ).into());
+        }
+
+        if last_cfg.raftdb.wal_dir != self.raftdb.wal_dir {
+            return Err(format!(
+                "raftdb wal_dir have been changed, former raftdb wal_dir is {}, \
+                 current raftdb wal_dir is {}, please guarantee all raft wal log \
+                 have been moved to destination directory.",
+                last_cfg.raftdb.wal_dir, self.rocksdb.wal_dir
+            ).into());
+        }
+
+        Ok(())
+    }
+
+    pub fn from_file<P: AsRef<Path>>(path: P) -> Self
```

Comment:
Can you replace https://github.com/pingcap/tikv/pull/2764/files#diff-49ffe9bfec96f94c952c9e3fa9985154R493 with this?